src/sentry/workflow_engine/models/data_source.py

```diff
@@ -10,12 +12,17 @@
 from sentry.workflow_engine.models.data_source_detector import DataSourceDetector
 
 
+class DataPacket(Protocol):
+    query_id: int
```

Comment:
not sure if this correct, i want to ignore all the other fields that might be on an object, and only enforce the query_id is set.